src/core/file_sys/directories/normal_directory.cpp

```diff
@@ -0,0 +1,139 @@
+
+namespace Core::Directories {
+
+std::shared_ptr<BaseDirectory> NormalDirectory::Create(std::string_view guest_directory) {
+    return std::static_pointer_cast<BaseDirectory>(
+        std::make_shared<NormalDirectory>(guest_directory));
+}
+
+NormalDirectory::NormalDirectory(std::string_view guest_directory) {
+    auto* mnt = Common::Singleton<Core::FileSys::MntPoints>::Instance();
+
+    static s32 fileno = 0;
+    mnt->IterateDirectory(guest_directory, [this](const auto& ent_path, const auto ent_is_file) {
+        auto& dirent = dirents.emplace_back();
+        dirent.d_fileno = ++fileno;
+        dirent.d_type = (ent_is_file ? 8 : 4);
+        strncpy(dirent.d_name, ent_path.filename().string().data(), MAX_LENGTH + 1);
+        dirent.d_namlen = ent_path.filename().string().size();
+
+        // Calculate the appropriate length for this dirent.
+        // Account for the null terminator in d_name too.
+        dirent.d_reclen = Common::AlignUp(sizeof(dirent.d_fileno) + sizeof(dirent.d_type) +
+                                              sizeof(dirent.d_namlen) + sizeof(dirent.d_reclen) +
+                                              (dirent.d_namlen + 1),
+                                          4);
+
+        directory_size += dirent.d_reclen;
+    });
+
+    // The last entry of a normal directory should have d_reclen covering the remaining data.
+    // Since the dirents of a folder are constant by this point, we can modify the last dirent
+    // before creating the emulated file buffer.
+    u64 filler_count = Common::AlignUp(directory_size, 512) - directory_size;
+    dirents[dirents.size() - 1].d_reclen += filler_count;
+
+    // Reading from standard directories seems to be based around file pointer logic.
+    // Keep an internal buffer representing the raw contents of this file descriptor,
+    // then emulate the various read functions with that.
+    directory_size = Common::AlignUp(directory_size, 512);
+    data_buffer.reserve(directory_size);
+    memset(data_buffer.data(), 0, directory_size);
+
+    u8* current_dirent = data_buffer.data();
+    for (NormalDirectoryDirent dirent : dirents) {
+        NormalDirectoryDirent* dirent_to_write =
+            reinterpret_cast<NormalDirectoryDirent*>(current_dirent);
+        dirent_to_write->d_fileno = dirent.d_fileno;
+
+        // Using size d_namlen + 1 to account for null terminator.
+        strncpy(dirent_to_write->d_name, dirent.d_name, dirent.d_namlen + 1);
+        dirent_to_write->d_namlen = dirent.d_namlen;
+        dirent_to_write->d_reclen = dirent.d_reclen;
+        dirent_to_write->d_type = dirent.d_type;
+
+        current_dirent += dirent.d_reclen;
+    }
+}
+
+s64 NormalDirectory::read(void* buf, u64 nbytes) {
+    // Nothing left to read.
+    if (file_offset >= directory_size) {
+        return ORBIS_OK;
+    }
+
+    s64 remaining_data = directory_size - file_offset;
+    s64 bytes = nbytes > remaining_data ? remaining_data : nbytes;
```

Comment:
```suggestion
    const s64 remaining_data = directory_size - file_offset;
    const s64 bytes = nbytes > remaining_data ? remaining_data : nbytes;
```
Also apply in other places where const is applicable